dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/JdbcAnalyticsManager.java

```diff
@@ -226,15 +228,60 @@ public void replaceDataPeriodsWithAggregationPeriods( Map<String, Object> dataVa
                 for ( DimensionalItemObject period : periods )
                 {
                     String[] keyCopy = keyArray.clone();
+
                     keyCopy[periodIndex] = ((Period) period).getIsoDate();
-                    dataValueMap.put( TextUtils.toString( keyCopy, DIMENSION_SEP ), value );
+
+                    String replacementKey = TextUtils.toString( keyCopy, DIMENSION_SEP );
+
+                    if ( dataValueMap.containsKey( replacementKey ) )
+                    {
+                        if ( requiresWeightedAvg( (Period) period ) )
+                        {
+                            value = calculateWeightedAvg( (Double) dataValueMap.get( replacementKey ), (Double) value,
+                                ((Period) period).getPeriodType() );
+
+                            dataValueMap.put( TextUtils.toString( keyCopy, DIMENSION_SEP ), value );
+                        }
+                    }
+                    else
+                    {
+                        dataValueMap.put( TextUtils.toString( keyCopy, DIMENSION_SEP ), value );
+                    }
                 }
 
                 dataValueMap.remove( key );
             }
         }
     }
 
+    private boolean requiresWeightedAvg( Period period  )
+    {
+        return period.getPeriodType().isFinancialYear();
+    }
+
+    private Double calculateWeightedAvg( Double year1Value, Double year2Value, PeriodType periodType )
+    {
+        int weight = calculateWeightFactor( periodType );
+
+        return Precision.round( (year1Value * ((double) (12 - weight) / 12)) + (year2Value * ((double) weight / 12)), 2 );
```

Comment:
Scale factor 2 is too small in general. Each indicator has a "precision" that determines how many digits after the decimal place are returned. If you have handy access to that precision, you could use it here. But if you do not... :) Numerators and Denominators will eventually be trimmed to this precision before sending them to the frontend -- see for example in DefaultAnalytics Service line 531:
```
.addValue( AnalyticsUtils.getRoundedValue( dataSourceParams, indicator.getDecimals(), value.getDenominatorValue() ) )
```
When defining an indicator, the UI allows you to select a precision of 0,1,2,3,4,5, or `<No value>`. If an indicator does not have a precision, a value of 10 digits is used, as you can see if you search for `DECIMALS_NO_ROUNDING` as defined in AnalyticsUtils. So if you round to 10 digits here I think that will handle all cases, and the value may be further rounded before it is sent to the frontend. I suggest a symbolic constant -- like you might make `DECIMALS_NO_ROUNDING` public and use it here.

Reply:
Ok, I went the simplest path and simply made `DECIMALS_NO_ROUNDING` public.